src/VirtoCommerce.MarketingModule.Data/Services/CouponService.cs

```diff
@@ -67,6 +66,13 @@ public async Task SaveCouponsAsync(Coupon[] coupons)
                         }
                         else
                         {
+                            var couponWithSameName = repository.Coupons
```

Comment:
It is better to move this logic for validation  into the repository and  receives the list of coupons and  returns the list of errors 
```C#
string[] CheckCouponsForUniqueness(Coupon[] coupons)
```

Reply:
done